### PR TITLE
feat: add support for auto-complete of classes in JARs

### LINKED ISSUE #7929: Adding a jar could scan it for class files

### PR DESCRIPTION
**Pull request by magnus-madsen**
Fixes #7929

### CODE REVIEW

**Comment by magnus-madsen** [comment]
image

**Comment by magnus-madsen** [comment]
Using the class files in the JAR is very fast approximation.

File: main/src/ca/uwaterloo/flix/api/Flix.scala [hunk 1]
```diff
@@ -678,4 +674,59 @@ class Flix {
...
+  }
+
+  /**
+    * Returns a multimap from Java packages to sub-packages, classes, and interfaces.
+    */
+  private def getPackageContent(l: List[String]): MultiMap[List[String], String] = {
+    l.foldLeft[MultiMap[List[String], String]](MultiMap.empty) {
+      case (acc, clazz) =>
+        // Given a string `java/util/zip/ZipUtils.class` we convert it to the list `java :: util :: zip :: ZipUtils`.
+        val clazzPath = clazz.stripSuffix(".class").split('/').toList
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/api/Flix.scala:716]: Does it work on Windows with the backslash?

**Review by mlutze — APPROVED** [review]

File: main/src/ca/uwaterloo/flix/api/Flix.scala (hunk 1 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/api/Flix.scala:716]: Its actually a zip file, so its /

**Pull request merged**